### ISSUE TITLE
How to avoid installing build dependency added by add_subdirectory?

### ISSUE THREAD

**Issue by Char-Aznable** [scikit-build/scikit-build#852]
Hi, I am using scikit-build to build and install a project with pure c++, c++ pybind11 and pure python code. When I run `pip install`, I find that the third-party libraries added via cmake's fetchcontent or add_subdirectory required to build the c++ and c++ pybind11 code are installed into my python site-packages directory.  For example, one of the dependency is a pure c++ header-only library that is only needed for building my python binding code but not needed for installation but somehow it's installed. Is there a way to avoid installing these third-party dependency? My setup.py looks like:
```python
from skbuild import setup
from setuptools import find_packages

if __name__ == "__main__":
    setup(
        cmake_languages=("CXX", "CUDA"),
        packages=find_packages(
            where=".",
            exclude=["*mypkg1*"],
            ),
        package_dir={"mymodule": "mymodule"},
        exclude_package_data={"mymodule" : ["*_deps*"]},
        cmake_install_dir="mymodule",
    )

```

**Comment by Char-Aznable** [comment]
Here is a minimal example. If I modify the example https://github.com/scikit-build/scikit-build-sample-projects/tree/master/projects/hello-pybind11 with this patch (basically just declaring and fetching a FetchContent dependency without eve n linking or using it): 
```diff
diff --git a/projects/hello-pybind11/CMakeLists.txt b/projects/hello-pybind11/CMakeLists.txt
index d57d774..6699598 100644
--- a/projects/hello-pybind11/CMakeLists.txt
+++ b/projects/hello-pybind11/CMakeLists.txt
@@ -13,6 +13,16 @@ FetchContent_Declare(
 )
 FetchContent_MakeAvailable(pybind11)
 
+FetchContent_Declare(
+  tinyDNN 
+  GIT_REPOSITORY https://github.com/tiny-dnn/tiny-dnn.git
+  GIT_TAG        origin/master
+  GIT_SHALLOW 1
+  GIT_PROGRESS ON
+)
+
+FetchContent_MakeAvailable(tinyDNN)
+
 set(python_module_name _hello)
 pybind11_add_module(${python_module_name} MODULE
   src/hello/hello_py.cpp

```
this is enough to trigger the installation of the entire `tiny-dnn` repo, which is undesirable.

**Comment by Char-Aznable** [comment]
Related to https://github.com/scikit-build/scikit-build/issues/500

**Comment by henryiii** [comment]
Files (I'd really not call them dependencies) are controlled by setuptoools. So you'll want a `MANIFEST.in` and you can exclude/include things there.

In scikit-build-core, you'd instead have include/exclude fields in the pyproject.toml.

**Comment by Char-Aznable** [comment]
@henryiii  Thanks for your response!  The third-party libraries I mentioned are actually dependencies to build my project, as opposed to what is shown in the aforementioned example, which I deliberately avoid adding the build dependency to show the problem. In any case, the problem is the third-party projects have their own cmake `install()` commands, which by default will be triggered to install into the site-packages root, which is undesirable. 

I then found the `cmake_install_target` option. It wasn't obvious from the documentation that "scikit-build will invoke cmake to install all add_subdirectory dependencies if not configured with cmake_install_target" is an thing to watch out for. 

Anyway, thanks for this great tool!

**Comment by henryiii** [comment]
scikit-build (classic) after #1185 is now a thin wrapper around scikit-build-core's setuptools plugin. Closing all open issues; if this still applies, please reopen or open an issue on scikit-build-core. Thanks!
